devops/jobs/AppWatcher.groovy

```diff
@@ -61,10 +61,11 @@ class AppWatcher {
 
         // for each application
         extraVars.get("APPS_TO_CONFIG").each { app_name, app_config ->
-            dslFactory.job(extraVars.get("FOLDER_NAME", "DockerCI") + "/" + app_name) {
+            dslFactory.job(extraVars.get("FOLDER_NAME", "DockerCI") + "/application-watchers/" + app_name + "-watcher") {
 
                 def app_repo = app_config.get("app_repo", '')
-                def app_repo_branch = app_config.get('app_repo_branch', 'master')
+                // if an application has no repository, the app_repo_branch should be blank so as not to set up webhooks
+                def app_repo_branch = !app_repo ? '' : app_config.get('app_repo_branch', 'master')
```

Comment:
Do we have anyone with an empty repo?  If so, we probably need to support polling, otherwise, manual and documented Works For Me